auth/identity_context.go

```diff
@@ -48,6 +51,14 @@ func (c IdentityContext) UserInfo() *service.UserInfoResponse {
 		return &service.UserInfoResponse{}
 	}
 
+	userInfo := c.userInfo
+	if c.claims != nil {
+		m, err := structpb.NewStruct(*c.claims)
+		if err != nil {
+			logger.Errorf(context.TODO(), "Failed to marshal claims [%+v] to struct: %v", c.claims, err)
+		}
+		userInfo.AdditionalClaims = m
+	}
```

Comment:
Can we move this to the constructor NewIdentityContext? no need to do this every time...

Reply:
Also, let's refactor the constructor to return an Error and don't silence the error from here?
Also, I think we can remove the error log as long as you augment the error you will return with the information